sdk/storage/Azure.Storage.Blobs.Batching/src/BlobBatchClient.cs

```diff
@@ -0,0 +1,774 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Storage.Blobs.Models;
+
+#pragma warning disable SA1402  // File may only contain a single type
+
+namespace Azure.Storage.Blobs.Specialized
+{
+    /// <summary>
+    /// The <see cref="BlobBatchClient"/> allows you to batch multiple Azure
+    /// Storage operations in a single request.
+    /// </summary>
+    public class BlobBatchClient
+    {
+        /// <summary>
+        /// Gets the blob service's primary <see cref="Uri"/> endpoint.
+        /// </summary>
+        public virtual Uri Uri { get; }
+
+        /// <summary>
+        /// The <see cref="HttpPipeline"/> transport pipeline used to send
+        /// every request.
+        /// </summary>
+        internal virtual HttpPipeline Pipeline { get; }
+
+        /// <summary>
+        /// The <see cref="HttpPipeline"/> transport pipeline used to prepare
+        /// requests for batching without actually sending them.
+        /// </summary>
+        internal virtual HttpPipeline BatchOperationPipeline { get; }
+
+        #region ctors
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BlobBatchClient"/>
+        /// class for mocking.
+        /// </summary>
+        protected BlobBatchClient()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BlobBatchClient"/>
+        /// class for the same account as the <see cref="BlobServiceClient"/>.
+        /// The new <see cref="BlobBatchClient"/> uses the same request policy
+        /// pipeline as the <see cref="BlobServiceClient"/>.
+        /// </summary>
+        /// <param name="client">The <see cref="BlobServiceClient"/>.</param>
+        public BlobBatchClient(BlobServiceClient client)
```

Comment:
Intentional that we don't have the same set of ctors that we have for the other clients?

Reply:
Yes - this is more like `LeaseClient` than `BlobClient`.